src/Agent.Worker/JobExtension.cs

```diff
@@ -131,17 +131,27 @@ public async Task<JobInitializeResult> InitializeJob(IExecutionContext jobContex
                     Stack<IStep> postJobStepsBuilder = new Stack<IStep>();
                     Dictionary<Guid, Variables> taskVariablesMapping = new Dictionary<Guid, Variables>();
 
-                    if (context.Container != null)
+                    if (context.Container != null || context.SidecarContainers?.Count > 0)
                     {
                         var containerProvider = HostContext.GetService<IContainerOperationProvider>();
-                        initResult.PreJobSteps.Add(new JobExtensionRunner(runAsync: containerProvider.StartContainerAsync,
+                        var containers = new List<Container.ContainerInfo>();
+                        if (context.Container != null)
+                        {
+                            containers.Add(context.Container);
+                        }
+                        if (context.SidecarContainers?.Count > 0)
+                        {
+                            containers.AddRange(context.SidecarContainers);
+                        }
+                        // TODO (@dakale): Localize strings
```

Comment:
Need to localize strings (required before merge?)

Reply:
@TingluoHuang It looks like in previous PRs, strings have been added to en-US, and then localized periodically. Is that okay for this type of change where the string will be user facing and not hidden in an error message?

Reply:
it's fine, btw, do we really need 2 loc string?
will `InitializeContainers` cover both case? :D